tuner: Allow specifying different tuner config file path

StephanDollberg opened the pull request:
We add cli flags to both `tune` and `start` to allow overriding the
tuner config file path.

This was requested by the k8s team for integration into the helm chart.

In that case it's safe to change the path to a non-tmpfs like path as
the tuner always reruns as part of the init-container before RP starts.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes


* none

StephanDollberg committed:
tuner: Allow specifying different tuner config file path

We add cli flags to both `tune` and `start` to allow overriding the
tuner config file path.

This was requested by the k8s team for integration into the helm chart.

In that case it's safe to change the path to a non-tmpfs like path as
the tuner always reruns as part of the init-container before RP starts.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--3c9d9742-8379-4ff3-9849-59bda9e0336e-->
<!--TEST RESULTS-->
<!--b3580002-97a0-4cba-85f8-43498692d7da-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74314">build#74314</a></summary>


| test_class                       | test_method                                  | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                              |
|:---------------------------------|:---------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkBasicTests             | test_rapid_shadow_link_toggling              | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74314#0199ec6b-ff58-4f73-afc3-5727b78b882b | FLAKY         | 16/21    | upstream reliability is '97.32142857142857'. current run reliability is '76.19047619047619'. drift is 21.13095 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_rapid_shadow_link_toggling        |
| SegmentMsTest                    | test_segment_rolling_with_retention_consumer | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74314#0199ec72-5e2c-4c0c-b811-07071d2fff13 | FLAKY         | 19/21    | upstream reliability is '90.30362389813908'. current run reliability is '90.47619047619048'. drift is -0.17257 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SegmentMsTest&test_method=test_segment_rolling_with_retention_consumer  |
| WriteCachingFailureInjectionTest | test_unavoidable_data_loss                   | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74314#0199ec6b-ff5d-427f-b6e8-e835931e244c | FLAKY         | 15/21    | upstream reliability is '90.72532699167658'. current run reliability is '71.42857142857143'. drift is 19.29676 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionTest&test_method=test_unavoidable_data_loss |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74412">build#74412</a></summary>


| test_class                    | test_method                                  | test_arguments                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   | test_history                                                                                                                                                               |
|:------------------------------|:---------------------------------------------|:----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_replication_basic                       | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "redpanda"}} | integration | https://buildkite.com/redpanda/redpanda/builds/74412#0199f2b7-1c1f-4f17-aea1-3a75da82883a | FLAKY         | 19/21    | upstream reliability is '96.60297239915074'. current run reliability is '90.47619047619048'. drift is 6.12678 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic         |
| ShadowLinkingReplicationTests | test_replication_with_failures               | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/74412#0199f2b7-1c20-49cb-b7fa-94bc5d54e14d | FLAKY         | 9/21     | upstream reliability is '85.99348534201955'. current run reliability is '42.857142857142854'. drift is 43.13634 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures |
| SegmentMsTest                 | test_segment_rolling_with_retention_consumer | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/74412#0199f2b0-942f-408c-8157-24204668def5 | FLAKY         | 20/21    | upstream reliability is '90.56956115779646'. current run reliability is '95.23809523809523'. drift is -4.66853 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SegmentMsTest&test_method=test_segment_rolling_with_retention_consumer   |
| QuickTerminateTest            | test_terminate                               | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/74412#0199f2b0-9428-4969-8469-d75340a411f6 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=QuickTerminateTest&test_method=test_terminate                            |
</details>
<!--/body-->

StephanDollberg committed:
tuner: Rename default node tuner state/config path

After some internal rethinking we decided to switch from config to state
as it's more output than input.

Also remove net from the path.
